packages/spindle-ui/src/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useCallback } from 'react';
+import MenuHorizontal from '../Icon/MenuHorizontal';
+
+import PaginationItem from './PaginationItem';
+import { useShowItem } from './hooks/useShowItem';
+
+interface Props extends React.HTMLAttributes<HTMLElement> {
+  current: number;
+  total: number;
+  showCount?: boolean;
+  showPrevNext?: boolean;
+  showFirstLast?: boolean;
+  onPageChange: (
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+    pageNumber: number,
+  ) => void;
+  createUrl: (pageNumber: number) => string;
+}
+
+const BLOCK_NAME = 'spui-Pagination';
+
+export const Pagination = (props: Props) => {
+  const {
+    current,
+    total,
+    showCount = false,
+    showPrevNext = true,
+    showFirstLast = false,
+    onPageChange,
+    createUrl,
+    className,
+    ...rest
+  } = props;
+
+  const pageItem = 5;
+  const { displayItem, showPrevHorizontal, showNextHorizontal } = useShowItem({
+    current,
+    total,
+    pageItem,
+  });
+
+  const handleOnClick = useCallback(
```

Comment:
nits: `handleClick` の方がより一般的・適切かもです (大きな問題ではないですが)

```suggestion
  const handleClick = useCallback(
```